src/plone/recipe/zope2instance/ctl.py

```diff
@@ -874,6 +879,43 @@ def help_adduser(self):
         print("adduser <name> <password> -- add a Zope management user")
 
 
+def serve_paste(app, global_conf, **kw):
+    sock = None
+    if 'prebound' in global_conf:
+        _sock = socket.fromfd(
+            int(global_conf['prebound']), socket.AF_INET, socket.SOCK_STREAM)
+        if six.PY2:
+            sock = socket.socket()
+            sock._sock = _sock
+        else:
+            sock = _sock
+        kw.update(sockets=[sock])
+    try:
+        serve(app, **kw)
+    finally:
+        if isinstance(sock, socket.socket):
+            sock.close()
+    return 0
+
+
+def server_factory(global_conf, **kws):
+    if 'fast-listen' in kws:
+        host, port = kws['fast-listen'].split(':')
+        prebound = dispatcher()
+        prebound.create_socket(socket.AF_INET, socket.SOCK_STREAM)
+        prebound.bind((host, int(port)))
+        prebound.listen(5)
+        while not prebound.readable():
+            sleep(1)
+        global_conf.update(prebound=str(prebound.socket.fileno()))
+        del kws['fast-listen']
+    del kws['paste.server_factory']
+
+    def serve(app):
+        return serve_paste(app, global_conf, **kws)
```

Comment:
Where does `global_conf` come from? It is meant to be a global variable?

Reply:
Sorry, overlooked the indentation. The name of the function overwrites the imported one from `waitress` maybe the name could be changed to prevent some confusion.